client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,772 @@
+import axe from 'axe-core';
+
+import { Controller } from '@hotwired/stimulus';
+import type { ContextObject } from 'axe-core';
+import {
+  getAxeConfiguration,
+  getA11yReport,
+  renderA11yResults,
+} from '../includes/a11y-result';
+import { wagtailPreviewPlugin } from '../includes/previewPlugin';
+import {
+  getPreviewContentMetrics,
+  renderContentMetrics,
+} from '../includes/contentMetrics';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+import { setOptionalInterval } from '../utils/interval';
+
+const runContentChecks = async () => {
+  axe.registerPlugin(wagtailPreviewPlugin);
+
+  const contentMetrics = await getPreviewContentMetrics({
+    targetElement: 'main, [role="main"], body',
+  });
+
+  renderContentMetrics({
+    wordCount: contentMetrics.wordCount,
+    readingTime: contentMetrics.readingTime,
+  });
+};
+
```

Comment:
Aside from adding TypeScript annotations and updating the `#preview-iframe` to `#w-preview-iframe` selector, I believe I didn't change anything to these two functions.